ciao-controller/internal/datastore/datastore.go

```diff
@@ -1107,14 +1104,17 @@ func (ds *Datastore) deleteInstance(instanceID string) (string, error) {
 		ds.nodesLock.Unlock()
 	}
 
-	err := ds.db.deleteInstance(i.ID)
-	if err != nil {
-		glog.V(2).Info("deleteInstance: ", err)
+	var err error
+	if tmpErr := ds.db.deleteInstance(i.ID); tmpErr != nil {
+		glog.Warningf("error deleting instance (%v): %v", i.ID, err)
+		err = errors.Wrapf(tmpErr, "error deleting instance from database (%v)", i.ID)
 	}
 
-	err = ds.ReleaseTenantIP(i.TenantID, i.IPAddress)
-	if err != nil {
-		glog.V(2).Info("deleteInstance: ", err)
+	if tmpErr := ds.ReleaseTenantIP(i.TenantID, i.IPAddress); tmpErr != nil {
+		glog.Warningf("error releasing IP for instance (%v): %v", i.ID, err)
+		if err == nil {
+			err = errors.Wrapf(err, "error releasing IP for instance (%v)", i.ID)
+		}
 	}
```

Comment:
Are the conditional wraps here required?  I understood from https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully examples that you could just wrap nil and the right thing happened.

And is keeping the glog useful?  I understood the idiom with errors.Wrap() to be one where you no longer do that.